alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -359,16 +361,16 @@ deriving via
 
 pattern WitnessPPData ::
   Era era =>
-  Map.Map RdmrPtr (Data era) ->
+  Redeemers era ->
   Set (LangDepView era) ->
   WitnessPPData era
-pattern WitnessPPData mp s <-
-  WitnessPPDataConstr (Memo (WitnessPPDataRaw mp s) _)
+pattern WitnessPPData r s <-
+  WitnessPPDataConstr (Memo (WitnessPPDataRaw r s) _)
   where
-    WitnessPPData mp s =
+    WitnessPPData r s =
       WitnessPPDataConstr
         . memoBytes
-        $ (Rec WitnessPPDataRaw !> mapEncode mp !> setEncode s)
+        $ (Rec WitnessPPDataRaw !> To r !> setEncode s)
```

Comment:
This is the encoding that we are going to have to explain in a document somewhere, and there are other choices we could go with as well. The important thing here is that we are at least using the user supplied encoding for the redeemers.

Reply:
In particular, the `setEncode` part